elixir/apps/domain/lib/domain/events/hooks/actor_group_memberships.ex

```diff
@@ -1,24 +1,38 @@
 defmodule Domain.Events.Hooks.ActorGroupMemberships do
-  alias Domain.{Events, Policies, PubSub}
+  alias Domain.{Events, Flows, Policies, PubSub, Repo}
 
   def on_insert(%{"actor_id" => actor_id, "group_id" => group_id} = _data) do
+    broadcast_access(:allow, actor_id, group_id)
     broadcast(:create, actor_id, group_id)
   end
 
   def on_update(_old_data, _data), do: :ok
 
   def on_delete(%{"actor_id" => actor_id, "group_id" => group_id} = _old_data) do
-    broadcast(:delete, actor_id, group_id)
+    Task.start(fn ->
+      {:ok, _flows} = Flows.expire_flows_for(actor_id, group_id)
+      broadcast_access(:reject, actor_id, group_id)
+      broadcast(:delete, actor_id, group_id)
+    end)
+
+    :ok
   end
 
   def broadcast(action, actor_id, group_id) do
     payload = {:"#{action}_membership", actor_id, group_id}
     topic = Events.Hooks.Actors.memberships_topic(actor_id)
 
     :ok = PubSub.broadcast(topic, payload)
+  end
 
-    # TODO: WAL
-    # This is an n+1 query; refactor with a cached lookup table on the client channel
-    :ok = Policies.broadcast_access_events_for(action, actor_id, group_id)
+  defp broadcast_access(action, actor_id, group_id) do
+    # TODO: There's likely a bug here - need to omit disabled policies too
```

Comment:
This was copied over from the prior implementation. Let me test it's not an issue real quick.